File: public/static/templates/python/Controller.py
forEach: Aggregate
fileName: {{namePascalCase}}Controller.py
path: {{boundedContext.name}}
---
from flask import Flask, abort, jsonify, request, Blueprint, Response
import {{namePascalCase}}Repository 
import Hateoas 

bp = Blueprint('{{nameCamelCase}}', __name__, url_prefix='/{{namePlural}}')

@bp.route("", methods=["POST"])
def Post():
    response = {{namePascalCase}}Repository.Create(request)

    response = Hateoas.POST_response(request.base_url, response)
    return response

@bp.route("", methods=["GET"])
def Get():
    response = {{namePascalCase}}Repository.Read(request)
    response = Hateoas.GET_list_response(request.url_root, request.base_url, response)
    
    return response

@bp.route("/<int:id>", methods=["GET"])
def Get_By_Id(id: int):
    response = {{namePascalCase}}Repository.Read_by_id(id)

    if response != 'error':
        response = Hateoas.GET_id_response(request.base_url, response)
    else:
        response = Response(response, status=404,mimetype='application/json')
    return response

@bp.route("/<int:id>", methods=["UPDATE"])
def Update(id: int):
    
    response = {{namePascalCase}}Repository.Update(id, request)
    
    if response != 'error':
        response = Hateoas.GET_id_response(request.base_url, response)
    else:
        response = Response(response, status=403,mimetype='application/json')

    
    return response

@bp.route("/<int:id>", methods=["DELETE"])
def Delete(id: int):
    response = {{namePascalCase}}Repository.Delete(id)
    if response == None:
        response = Response(status=200, mimetype='application/json')
    else :
        response = Response(response, status=403,mimetype='application/json')
    
    return response

{{#commands}}
{{#isRestRepository}}
{{/isRestRepository}}
{{^isRestRepository}}
@bp.route("/{{controllerInfo.apiPath}}", methods=["{{controllerInfo.method}}"])
def {{nameCamelCase}}(): # -> 이부분 수정 
    
    '''
    LOGIC GOES HERE
    '''
    return 'hello world!'

{{/isRestRepository}}
{{/commands}}

@bp.errorhandler(404)
def resource_not_found(e):
    return jsonify(error=str(e)), 404

File: public/static/templates/nodejs/routes/controller.js
forEach: Aggregate
fileName: {{namePlural}}.js
path: {{boundedContext.name}}/routes
---
var express = require('express');
var router = express.Router();
var repository = require('../repository/{{namePascalCase}}Repository');

var name = "{{nameCamelCase}}";

router.post('/', function(req, res, next) {
  
  repository.save(req.body).then(result => {
    res.json(hateoas.hal_post(result.dataValues, req, name));
  })
  

});

router.get('/', function(req,res,next){
  repository.read().then(result => {
    res.json(hateoas.hal_get_list(result, req, name));
  });
});

router.get('/:id', function(req,res,next){
  repository.read_by_id(req.params.id).then(result=>{
    res.json(hateoas.hal_get_id(result, req, name));
  });
});

router.put('/:id', function(req,res,next){
  repository.update(req.params.id, req.body).then(result=>{
    res.json(hateoas.hal_get_id(result, req, name));
  });
})

router.delete('/:id', function(req,res,next){
  repository.delete(req.params.id).then(result=> {
    res.json(result)
  });
})

{{#commands}}
{{^isRestRepository}}
router.{{#methodCheck controllerInfo.method}}{{/methodCheck}}('/{{controllerInfo.apiPath}}', function(req, res, next){
  // LOGIC GOES HERE
  return "COMMAND TYPE"
})

{{/isRestRepository}}
{{/commands}}

module.exports = router;

<function> 
  window.$HandleBars.registerHelper('methodCheck', function (method) {
      if(method.endsWith("POST")){
          return "post"
      }
      else if(method.endsWith("GET")){
          return "get"
      }
      else if(method.endsWith("UPDATE")){
        return "put"
      }
      else if(method.endsWith("DELETE")){
        return "destroy"
      }
  });
</function>
